Replace pep257 by pydocstyle

airvzxf opened the pull request:
Regarding the issue #457, the `pep257` was replaced by `pydocstyle`, and fixed some warnings/errors.

TheLastProject commented on tox.ini:
```diff
@@ -3,22 +3,18 @@
 
-[tox]
-# 3.5 is the lowest supported versions
-envlist = py35, py36, py37
```
I need to look up what this actually changes, but it's past midnight. So I'll look this up tomorrow :)

airvzxf replied:
WoW! I assume that you are from the USA but looking at your profile and website, you are from the Netherlands.

Well, I was tentative to keep this without changes, but it is not really using because in the GitHub action `main.yml` / `build-linux.sh` you are specifying the version of Python `xvfb-run tox -v -e py"${PYTHON_VERSION//./}"` which is replaced by `xvfb-run tox -v -e py39`. Same case for macOS.

Furthermore, if you try to keep this `envlist` versions, they are not working because looks that you need to install via Tox or Ubuntu/macOS the different versions of Python. If you try to follow command `xvfb-run tox -v`, it will crash because it didn't find the Python versions. As, I showed in this screenshot:

![01-PythonVersions](https://user-images.githubusercontent.com/831380/198748703-acade103-2993-4daa-b24f-e0006ee36447.png)

TheLastProject replied:
Makes sense! Thanks!

TheLastProject commented on pext/ui_qt5.py:
```diff
@@ -1557,7 +1557,7 @@ def __init__(self, main_loop_queue: Queue, window: 'Window') -> None:
 
     def on_press(self, key) -> bool:
-        """Executed when a key is pressed."""
+        """Execute when a key is pressed."""
```
This seems like bad grammar now. The old description was quite unclear anyway. Maybe something like this:

```suggestion
        """Track pressed keys and show window when all keys for the global hotkey are pressed."""
```

airvzxf replied:
Fixed in the commit https://github.com/airvzxf/Pext/commit/f765d42edac30a6f25f3e53c0791c76b7c61e31f.

TheLastProject commented on pext/ui_qt5.py:
```diff
@@ -1572,7 +1572,7 @@ def on_press(self, key) -> bool:
 
     def on_release(self, key) -> bool:
-        """Executed when a key is released."""
+        """Execute when a key is released."""
```
```suggestion
        """Remove key from pressed keys list on release."""
```

airvzxf replied:
Fixed in the commit https://github.com/airvzxf/Pext/commit/f765d42edac30a6f25f3e53c0791c76b7c61e31f.

TheLastProject reviewed (changes requested):
Looks good. I think we'll want to use the opportunity of having to fix these comments to improve them as well :)

airvzxf commented:
**Please don't merge.** I'll address the problem for Python 3.10 and check if Python 3.11 is available in the GitHub images.

airvzxf commented:
Question, has the GitHub consumed your actions minute quota? I thought it would consume mine since they run on my repository airvzxf/pext, but I'm thinking it's consuming yours.

To review you can go to your account **settings**, then **bills and plans**, and the **usage this month**, there is the quota.

![image](https://user-images.githubusercontent.com/831380/198841329-62ed4ef0-10e9-4793-9312-552973b985b2.png)

![image](https://user-images.githubusercontent.com/831380/198841344-56e2b79c-af08-4798-a6fb-42e72c57523c.png)

TheLastProject commented:
It's all at 0 for me too, I guess Open Source projects just don't count maybe

airvzxf commented:
Good news! The change from pep257 to pydocstyle addressed the issues about imports of collections. It means your project is running for Python 3.7, 3.8, 3.9, 3.10, and 3.11.

Packaging action:
https://github.com/airvzxf/Pext/actions/runs/3352039425

Furthermore, if you want to test the artifacts, feel free. But, they should be work fine.

Please, told me if you are agreed with these changes to commit the last changes, which includes uncomment the upload job in the packaging action.

TheLastProject commented:
Sorry for the delay, forgot about this one amid all the chaos (lot of people seem to be trying to finish their Hacktoberfest the last week). I've labelled it hacktoberfest-accepted for now just in case :)

> What Python version do you want to keep in the packaging action?

Let's go for whatever is the most recent stable that we can build on Ubuntu 18.04. I don't have much time to support every possible option, so just supporting 3.11 seems best :)

> Please, told me if you are agreed with these changes to commit the last changes, which includes uncomment the upload job in the packaging action.

You shouldn't have to comment the upload job ever, pyuploadtool should be smart enough to not try to upload anything on a PR, so feel free to uncomment it.

It's looking quite awesome :)

airvzxf commented:
All the goals and requests are done. Ready to merge!

This pull request marks “Changes requested”, but they were fixed.

The artifacts are in https://github.com/Pext/Pext/actions/runs/3362695008.

TheLastProject commented:
Thank you!
